pkg/pillar/cmd/zedmanager/handlevolumemgr.go

```diff
@@ -401,10 +401,21 @@ func restoreConfigFromSnapshot(ctx *zedmanagerContext, appInstanceStatus *types.
 		return nil, fmt.Errorf("SnapshotInstanceStatus not found for %s", snapshotID)
 	}
 	// Get the app instance config from the snapshot
-	snappedAppInstanceConfig := deserializeConfigFromSnapshot(snapshotStatus)
+	snappedAppInstanceConfig := deserializeConfigFromSnapshot(snapshotStatus.Snapshot.SnapshotID)
 	if snappedAppInstanceConfig == nil {
 		return nil, fmt.Errorf("failed to read AppInstanceConfig from file for %s", snapshotID)
 	}
+	config, err := addFixupsIntoSnappedConfig(ctx, appInstanceStatus, snappedAppInstanceConfig)
+	if err != nil {
+		return config, err
+	}
+	return snappedAppInstanceConfig, nil
+}
+
+// addFixupsIntoSnappedConfig adds the fixups into the snapped app instance config.
+// The fixups are the information that should be taken from the current app instance config, not from the snapshot.
+// The fixups should correspond to the fixups done on the cloud side, in a function like: applyAppInstanceFromSnapshot
```

Comment:
s/cloud/controller/ to keep this generic, and I don't know if it makes sense to referring to a particular function in a particular controller here.

Reply:
When mentioning the function name, it's only meant as a hint for those who will be revising or debugging this section later on. Hence, I phrased it as "a function _like_ ..." to provide an idea without being too specific. But I can remove it if you find it inappropriate.

Reply:
Reworked the comment in the code.